browser/pFa.go

```diff
@@ -0,0 +1,139 @@
+package browser
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"log"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/vertgenlab/gonomics/fasta/pFasta"
+	"github.com/vertgenlab/gonomics/fileio"
+)
+
+// PFaVisualizer produces command line visualizations of pFasta format alignments from a specified start and end position.
+// Can be written to a file or to standard out. Includes noMask and lineLength formatting options as bools.
+// If 0 sig figs, returns full probability
+// There will be slight floating point errors in the last (or last two) places
+func PFaVisualizer(infile string, outfile string, start int, end int, endOfAlignment bool, sigFigs int, decimalPlaces int, lineLength int, seqName string) {
+	if !endOfAlignment && !(start < end) {
+		log.Fatalf("Error: Invalid arguments, start must be lower than end")
+	}
+
+	records := pFasta.Read(infile)
+	out := fileio.EasyCreate(outfile)
+
+	var err error
+	var formatting string
+	var formatNum int
+	if sigFigs == 0 {
+		formatting = "Decimal Places"
+		formatNum = decimalPlaces
+	} else {
+		formatting = "SigFigs"
+		formatNum = sigFigs
+	}
+	if !endOfAlignment {
+		_, err = fmt.Fprintf(out, "Start: %d. End: %d. %s: %d.", start, end, formatting, formatNum)
+		exception.PanicOnErr(err)
+	} else {
+		_, err = fmt.Fprintf(out, "Start: %d. End: to end. %s: %d.", start, formatting, formatNum)
+		exception.PanicOnErr(err)
+	}
+
+	if seqName == "" {
+		// only accepts single pFa or a .pFa with multiple pFastas and a specified record name
+		if len(records) > 0 {
```

Comment:
if the pfasta file is empty, would len(records) be ==0?

Reply:
Yes, I think the best thing to do if len(records)==0 would be to give a log.FatalF. I'll try to add that in a commit this week.